lib/Activity.py

```diff
@@ -817,7 +829,53 @@ def watch_next_wf(self, sessionid):
             outfile = os.path.join(sessions_dir, f"{wfid}-{stage}.yaml")
             with open(outfile, "w", encoding='UTF-8') as fhandle:
                 yaml.dump(cfgmaps, fhandle)
+            files = [file for file in os.listdir(self.media_dir) if file.endswith(".tar.gz")]
+            try:
+                cfgmap_products = ["{}-{}".format(p["name"], p["version"]) for p in cfgmaps["products"]]
+            except TypeError:
+                # This happens during a new activity on process-media.
+                cfgmap_products = {}
+
+            # Check the integrity of the tarballs in media_dir.
+            for product in files:
+                prodpath = os.path.join(self.media_dir, product)
+                tar_paths = []
+                try:
+                    with tarfile.open(prodpath, "r:gz") as tarf:
+                        counter = 0
+                        for entry in tarf:
+                            tar_paths.append(entry.name)
+                            counter += 1
+                            if counter > 10:
+                                break
+                except tarfile.ReadError as readerr:
+                    self.config.logger.debug(f"Problems extracting {prodpath}:")
+                    self.config.logger.debug(f"\t{readerr}")
+                    self.config.logger.debug("Ignoring it!")
+                    continue
+                tar_path = None
+                for path in tar_paths:
+                    if path.startswith('.'):
+                        continue
+                    tar_path = os.path.join(self.media_dir, path.split("/")[0])
+                    break
+                if not tar_path:
+                    # This should only be hit with a corrupt or empty tarball.
+                    self.config.logger.debug(f"Couldn't find a directory for the {prodpath} tar file")
+                    continue
 
+                if tar_path and os.path.exists(tar_path):
+                    # The path exists and process_media has been run.  Note
+                    # the directory will not exist before process-media is ran.
+                    dir_contents = os.listdir(tar_path)
+                    if "iuf-product-manifest.yaml" in dir_contents:
+                        matches = [cp for cp in cfgmap_products if cp in product]
+                        if len(matches) < 1 and stage != "process-media":
+                            self.config.logger.warning(f"It looks like process-media hasn't been ran since {product} was added to the media directory")
```

Comment:
How about "Tar file {tar_path} found in media directory but is not present in the list of products IUF is working with, you will need re-run starting from the process-media stage to include this product."